sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -70,116 +53,82 @@ public DefaultService(File pathToYamlSpecification) throws Exception {
         try {
             curatorMutex.release();
         } catch (Exception ex) {
             LOGGER.error("Error releasing ZK lock.", ex);
         }
     }
 
-    private void initCurator() {
-        curatorClient = CuratorFrameworkFactory.newClient(zkConnectionString, CuratorUtils.getDefaultRetry());
-        curatorClient.start();
-    }
-
-    protected void init() {
-        this.zkConnectionString = this.serviceSpec.getZookeeperConnection();
-        initCurator();
-        lock();
-
-        this.apiPort = this.serviceSpec.getApiPort();
-        this.stateStore = DefaultScheduler.createStateStore(this.serviceSpec, zkConnectionString);
-
-        try {
-            configTargetStore = DefaultScheduler.createConfigStore(serviceSpec, zkConnectionString, Arrays.asList());
-        } catch (ConfigStoreException e) {
-            LOGGER.error("Unable to create config store", e);
-            throw new IllegalStateException(e);
-        }
-
-        ConfigurationUpdater.UpdateResult configUpdateResult = DefaultScheduler
-                .updateConfig(serviceSpec, stateStore, configTargetStore);
-
-        this.targetConfigId = configUpdateResult.targetId;
-        offerRequirementProvider = DefaultScheduler.createOfferRequirementProvider(stateStore, targetConfigId);
-    }
-
-    @VisibleForTesting
-    protected Collection<Plan> generatePlansFromRawSpec(RawServiceSpecification rawServiceSpecification)
-            throws Exception {
-        DefaultPlanGenerator planGenerator = new DefaultPlanGenerator(configTargetStore, stateStore);
-        List<Plan> plans = new LinkedList<>();
-        if (rawServiceSpecification.getPlans() != null) {
-            for (Map.Entry<String, RawPlan> entry : rawServiceSpecification.getPlans().entrySet()) {
-                plans.add(planGenerator.generate(entry.getValue(), entry.getKey(), serviceSpec.getPods()));
-            }
-        }
-        return plans;
-    }
-
     /**
-     * Creates and registers the service with Mesos, while starting a Jetty HTTP API service on the
-     * {@code apiPort}.
+     * Creates and registers the service with Mesos, while starting a Jetty HTTP API service on the {@code apiPort}.
      */
     @Override
-    public void register(ServiceSpec serviceSpecification, Collection<Plan> plans) {
-        DefaultScheduler defaultScheduler = DefaultScheduler.create(
-                serviceSpecification,
-                plans,
-                stateStore,
-                configTargetStore,
-                offerRequirementProvider);
-
-        startApiServer(defaultScheduler, apiPort);
-        registerFramework(defaultScheduler, getFrameworkInfo(), "zk://" + zkConnectionString + "/mesos");
-    }
-
-    public ServiceSpec getServiceSpec() {
-        return serviceSpec;
-    }
+    public void register() throws Exception {
+        DefaultScheduler defaultScheduler = schedulerBuilder.build();
+        ServiceSpec serviceSpec = schedulerBuilder.getServiceSpec();
```

Comment:
The builder.build() call does some extra work internally, specifically creating the statestore/configstore (if not manually provided by the developer) and then validating/updating the target config. Figured it'd make sense to keep that work out of the constructor as it's effectively network access.